src/trace/subscriber.rs

```diff
@@ -0,0 +1,92 @@
+use tracing::*;
+use serde_derive::{Serialize, Deserialize};
+use std::{fmt, fs::File, io::Write, sync::Mutex};
+
+pub struct JsonLogger {
+    log_file: Mutex<File>,
+}
+
+impl JsonLogger {
+    pub fn new(log_file: File) -> JsonLogger {
+        JsonLogger {
+            log_file: Mutex::new(log_file),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ModelRecord {
+    event: Option<String>,
+    model_id: Option<u64>,
+    parent: Option<u64>,
+    model: Option<String>,
+}
+
+impl<'a> ModelRecord {
```

Comment:
I didn't see the usage of 'a in this struct?

Reply:
ah nice catch- I went back and forth on the type of the struct but didn't cleanup properly.